algosdk/abi/tuple_type.py

```diff
@@ -128,7 +130,7 @@ def _compress_multiple_bool(value_list):
                 result |= 1 << (7 - i)
         return result
 
-    def encode(self, values):
+    def encode(self, values: list) -> bytes:
```

Comment:
Is this `encode` method more restrictive than the others? I.e., it doesn't accept `bytes` or `bytearray` ?

Reply:
For tuples, `encode()` can basically take a list of any `ABIType`, so if I had a tuple like `(bool,byte,uint8)`, it would take a list like: `[True, b"\x01", 24]` and encode that into `bytes` so we can send it off to an ABI compliant app. It could take `bytes` object inside the list if needed. Similarly, `decode()` should convert the `bytes` or `bytearray` back to the list of `ABITypes`.

Reply:
so maybe `values: List[ABIType]) ...`. ?

Reply:
My bad, it should be list of any "ABIValues". We didn't define that yet so I'm incline to use a `List[Any]` for this PR.

Reply:
Actually good catch, I forgot that `TupleType` can accept `bytes` or `bytearray` if the child types are `ByteType` -- will fix this!